### ISSUE TITLE
Suspend subscriptions of users who don't check messages

### ISSUE THREAD

**Issue by pleary** [inaturalist/inaturalist#758]
I've been thinking about ways to improve Updates and I think this is something worth pursuing. There's no need to be using our resources generating data for people that don't use it.

Using a dump from 10/10/2015, there are 33.4 million updates for 41,836 subscribers. 34,627 of those subscribers have not viewed any of their message we have on record (we are now keeping 3 months of updates, so they have not read any within the last 3 months). Those users account for 20.5 of the 33.4 million updates.

I'm proposing if we find someone hasn't read any messages in the last 3 months then we suspend their subscriptions (we can probably even keep their existing updates around since they will be deleted in another 3 months). If we find they come back and read messages, their subscriptions will be reenabled. 

This could lead to generating 61% fewer updates, which will have all the benefits that go along with that (fewer and faster Updates delayed_jobs, less load on hevea, less storage in DB, faster dashboards, etc).

**Comment by kueda** [comment]
Sounds great to me.

**Comment by carrieseltzer** [comment]
Wow! 4/5 of people aren't opening their updates within 3 months? I wonder
if you'd have any success with a "[name], have you checked iNaturalist
lately?" email going out to all of those folks (or anyone who reaches the 3
months without checking in mark). Just a thought to try and re-engage some
of them.

Carrie

On Sun, Oct 25, 2015 at 7:22 PM, Ken-ichi <notifications@github.com> wrote:

> Sounds great to me.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/inaturalist/inaturalist/issues/758#issuecomment-150986881>
> .
>



-- 
Carrie E. Seltzer, Ph.D.
National Geographic's Great Nature Project
www.greatnatureproject.org

**Comment by naturalistcharlie** [comment]
there are so many people who vanish... It's part of why community ID was
added, right? You can usually tell... someone who added stuff over a short
period of time and then stop. Often it is part of a class they were
required to do but there is no requirement to follow up and curate their
data.

On a more upbeat note, while most PEOPLE end up lapsing, most OBSERVATIONS
seem to be attached to people who are dedicated users and stick around.

The casual vs 'hardcore' user balance is already an odd mix, it would be
great if people stuck around more, i think the issue of class projects and
such is different from other users who arrive and then leave though.

On Sun, Oct 25, 2015 at 7:57 PM, carrieseltzer <notifications@github.com>
wrote:

> Wow! 4/5 of people aren't opening their updates within 3 months? I wonder
> if you'd have any success with a "[name], have you checked iNaturalist
> lately?" email going out to all of those folks (or anyone who reaches the 3
> months without checking in mark). Just a thought to try and re-engage some
> of them.
>
> Carrie
>
> On Sun, Oct 25, 2015 at 7:22 PM, Ken-ichi <notifications@github.com>
> wrote:
>
> > Sounds great to me.
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/inaturalist/inaturalist/issues/758#issuecomment-150986881
> >
> > .
> >
>
>
>
> --
> Carrie E. Seltzer, Ph.D.
> National Geographic's Great Nature Project
> www.greatnatureproject.org
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/inaturalist/inaturalist/issues/758#issuecomment-150989034>
> .
>



-- 
============================
Charlie Hohn
Montpelier, Vermont

**Comment by alexshepard** [comment]
Is there any way to tell how many are simply gone, vs those who didn't check their updates but did open the web site or a mobile app in the intervening 3 months?

**Comment by tnywlls** [comment]
These are the updates pushed out as emails? How would you know if they were read?  Some of us have security/privacy settings that don't open remote images etc in emails.

**Comment by naturalistcharlie** [comment]
It seems what is more important is if they look at the website. They could
lovingly read each email but if they are never coming back to the site it
doesn't really matter.

On Tue, Oct 27, 2015 at 4:46 AM, tnywlls <notifications@github.com> wrote:

> These are the updates pushed out as emails? How would you know if they
> were read? Some of us have security/privacy settings that don't open remote
> images etc in emails.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/inaturalist/inaturalist/issues/758#issuecomment-151415700>
> .
>



-- 
============================
Charlie Hohn
Montpelier, Vermont

**Comment by pleary** [comment]
@tnywlls yes these are what users get in emails, and on the site and apps in the form of updates on things they follow - not updates about actions taken on their own content. So they would still get ID notifications for example. What is being suggested is if they don't log into the site or use any of the apps for 3 months, we suspend their subscriptions. We can't monitor if they read their emails, so if someone is just reading emails for 3 months and never using the site or apps, their subscriptions would be suspended too.

Here is what I recommend be done:
- [ ] add `last_active` (Date) and `subscriptions_suspended_at` (Datestamp) fields to `Users`
- [ ] add `ApplicationController` method to set `users.last_active = Time.now` and `users.subscriptions_suspended_at = nil` whenever there is a `current_user`
- [ ] add rake task, or append logic to `inaturalist:delete_expired_updates`, which sets `user.subscriptions_suspended_at = Time.now` if a user has updates, but all `updates.viewed_at` are NULL
- [ ] add logic to has_subscriptions which prevents updates from being generated for users with suspended updates

We could also have it send out a one-time email to the user when we set `user.subscriptions_suspended_at`. Something to the effect of, "You've been away and we stopped your subscriptions. Login to the site or use one of the mobile apps to resume". Does this seem reasonable? What to people think about sending an email when we suspend subscriptions?

**Comment by md5** [comment]
@pleary It might be good to send out an email a couple/few days before the subscriptions are suspended and another when they actually are suspended.

**Comment by pleary** [comment]
The above commit handles all the tasks I created. There is nothing yet that will email or notify anyone. Anyone have more thoughts on if/how/when we should notify people?

**Comment by kueda** [comment]
IMO, it's probably worth sending them an email when they are suspended, but not before that. I think the email should say something like, "It looks like you haven't been visiting the site so we suspended update notifications for you. If you'd like to re-activate them, click here" and there should be a link to unsuspend updates for that user.

**Comment by pleary** [comment]
Initially there will be over 35,000 people we will be suspending updates for, as they haven't viewed any of their updates, which we keep for 3 months. Should we send emails to the initial 35,000 people? If so can we just use the Rails mailer or will that need something else than can handle the scale?

How about this for the email text:
**Title**: [iNaturalist.org] Your subscriptions are on hold
**Body**: Hi User,
  You have been inactive on iNaturalist.org for a while, so we have suspended your subscriptions. To resume getting updates on site activity, please [log in to your account](http://www.inaturalist.org/login) again.

**Comment by kueda** [comment]
I think a Rails mailer is probably appropriate, since this is something we'll theoretically be doing on a semi-regular basis. Text looks good to me.

**Comment by kfc35** [comment]
It seems that the logic to set `user.subscriptions_suspended_at = Time.now` and to send the message in the rake task had been commented out in 2016 (https://github.com/inaturalist/inaturalist/commit/58b360365b7c0d4173b1705f80963bde33c68c36#diff-5d66b933382605f340f8c5bdff1c3c105d3a7681f663626d07c870968ee36720) and then finally deleted in 2024 (https://github.com/inaturalist/inaturalist/commit/bf63ae6b92a7fffc75baf956c52afdead55cbd08#diff-5d66b933382605f340f8c5bdff1c3c105d3a7681f663626d07c870968ee36720)

Is that logic still desired? Just want to make sure if the deletion was an intentional decision. This issue can be closed if the logic is no longer desired.
